src/dlangui/platforms/sdl/sdlapp.d

```diff
@@ -64,12 +63,22 @@ private derelict.util.exception.ShouldThrow missingSymFunc( string symName ) {
                 "SDL_RemoveTimer", "SDL_RemoveTimer", "SDL_PushEvent",
                 "SDL_RegisterEvents", "SDL_WaitEvent", "SDL_StartTextInput",
                 "SDL_Quit", "SDL_HasClipboardText", "SDL_GetClipboardText",
-                "SDL_free", "SDL_SetClipboardText", "SDL_Init", "SDL_GetNumVideoDisplays"]) {//"SDL_GetDisplayDPI"
-        if (symName.equal(s)) // Symbol is used
-            return derelict.util.exception.ShouldThrow.Yes;
+                "SDL_free", "SDL_SetClipboardText", "SDL_Init", "SDL_GetNumVideoDisplays"]; //"SDL_GetDisplayDPI"
+    foreach(info; errors)
+    {
+        import std.array;
+        import std.algorithm;
+        import std.exception;
+        import core.stdc.string;
+        // NOTE: this has crappy complexity as it was just updated as is
+        //     it also does not checks if the symbol was actually loaded
+        auto errMsg = cast(string) info.message[0 .. info.message.strlen];
+        bool found = names
```

Comment:
I guess this code is wrong because it finds `errMsg` in `names` instead of `names` in `errMsg`. Also using `array` to count items is a bad practice.
```suggestion
    bool found = any!(s=>errMsg1.canFind(s))(names);
```

Reply:
that's why there is a note, it is definitely ugly but this code is basically just run once on startup, it doesn't do much harm, so it can be improved later

Reply:
It's not just ugly, it's wrong, OTOH it is not related to the PR itself

Reply:
Yep, I agree it sucks, but the overall logic still valid, the problem is that it searches for substring in array of strings - it doesn't matter whether it searches name in errors or errors for name, in any case it is MxN task, but indeed as it was updated just as is there also another loop which makes it even worse. 

It will throw exception if some symbols were unable to be loaded, another option would be to simply log an error and let it crash on null pointer call at some stage later which I think is far more frustrating to both programmer and a user.

I'll see what I can do to improve it.

Reply:
Well, as I understand it does matter whether it searches name in errors or errors for name - definitely you will always fail if you try to find the whole error message in a set of function names just because the error message is longer than some function name and contain other words than a function name.